### ISSUE TITLE
useQuery not updating loading state

### ISSUE THREAD

**Issue by samrith-s** [apollographql/apollo-client#6253]
**Intended outcome:**

```javascript
const { data, loading, error } = useQuery(HELLO_WORLD_QUERY);
```
Once the query is complete it `loading` should be `false`.

**Actual outcome:**

The `loading` state is always `true`. This results in the page loading infinitely.

**How to reproduce the issue:**

- Add any query.
- Use `useQuery` from `@apollo/client@^3.0.0-beta.44`.
- Log `loading`.

**Versions**

**System:**
- OS: macOS Mojave 10.14.5
  
**Binaries:**
- Node: 10.16.3 - ~/.nvm/versions/node/v10.16.3/bin/node
- Yarn: 1.22.4 - ~/Work/troup/client/node_modules/.bin/yarn
- npm: 6.9.0 - ~/.nvm/versions/node/v10.16.3/bin/npm

**Browsers:**
- Edge: 81.0.416.68
- Safari: 12.1.1
  
**npmPackages:**
- @apollo/client: ^3.0.0-beta.44 => 3.0.0-beta.44
- apollo: ^2.27.3 => 2.27.3
- apollo-client: ^2.6.8 => 2.6.8
- react-apollo: ^3.1.5 => 3.1.5

**Comment by benjamn** [comment]
Try `beta.46` (just released), which includes PR #6221?

**Comment by StinsonZhao** [comment]
The same problem when using `beta.43`... T_T

```javascript
const { data, loading, error, networkStatus } = useQuery(MY_CERTAIN_QUERY)
// loading is aways true
// networkStatus always equals to 1
```


Then, I tried `beta.46` as @benjamn said above. Emm... **New Problem**:

- `networkStatus` can change from 1 to 7, that means the `loading` can change to false
- **But, the `data` is `undefined`(`error` is also `undefined`)**

I'm very sure that my QUERY is correct. The Network of Chrome DevTools shows the request was returned with data correctly.

By the way, not all of my queries can meet this problem, just some of them, but I cannot know why and how to fix it.

**Comment by samrith-s** [comment]
@benjamn The loading works. But there is a new issue with `useLazyQuery` and `error`. I will create a new issue for it.

**Comment by hwillson** [comment]
It looks like the original issue was fixed by `@apollo/client@3.0.0-beta.46`, so closing this off. Thanks!
